### PR TITLE
Remove deprecated Jetty plugin

### PR DESCRIPTION
**Pull request by eriwen**
### Context
Plugin has been deprecated. This PR removes it entirely and only leaves a note in the jetty plugin doc that it has been removed (for those who have it bookmarked, otherwise we don't link to it).

- jetty/src/main/java/org/gradle/api/plugins/jetty/AbstractJettyRunTask.java
- jetty/src/main/java/org/gradle/api/plugins/jetty/JettyPlugin.java
- jetty/src/main/java/org/gradle/api/plugins/jetty/JettyPluginConvention.java
- jetty/src/main/java/org/gradle/api/plugins/jetty/JettyRun.java
- jetty/src/main/java/org/gradle/api/plugins/jetty/JettyRunWar.java
- jetty/src/main/java/org/gradle/api/plugins/jetty/JettyStop.java
- jetty/src/main/java/org/gradle/api/plugins/jetty/ScanTargetPattern.java

Examples have been replaced with simple use of Gretty plugin where possible.

### Contributor Checklist
- [X] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [X] Ensure that tests pass locally: `./gradlew quickCheck <impacted-subproject>:check`

[CI Build](https://builds.gradle.org/viewQueued.html?itemId=2641170&tab=queuedBuildOverviewTab)

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation including proper use of `@since` and `@Incubating` annotations for all public APIs
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/docs/src/samples/webApplication/customized/build.gradle [hunk 1]
```diff
@@ -1,4 +1,13 @@
-apply plugin: 'jetty'
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        classpath 'org.akhikhl.gretty:gretty:1.4.2'
```
Comment (bmuschko) [inline subprojects/docs/src/samples/webApplication/customized/build.gradle]: I am bit concerned that the version of the plugin gets stale. Could we instead just link to the web application getting started guide we are currently writing? https://github.com/gradle/guides/issues/75

File: subprojects/docs/src/samples/webApplication/quickstart/build.gradle [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
-    mavenCentral()
+    jcenter()
 }
 
 dependencies {
     compile group: 'commons-io', name: 'commons-io', version: '1.4'
     compile group: 'log4j', name: 'log4j', version: '1.2.15', ext: 'jar'
 }
-
-httpPort = 8080
```
Comment (bmuschko) [inline subprojects/docs/src/samples/webApplication/quickstart/build.gradle]: Looks like we also want to show the configuration of the HTTP port. Something we can configure in Gretty as well.

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/samples/SamplesWebProjectIntegrationTest.groovy [hunk 3]
```diff
@@ -53,60 +52,6 @@ class SamplesWebProjectIntegrationTest extends AbstractIntegrationSpec {
                 'webapp.html')
     }
 
-    @LeaksFileHandles
```
Comment (bmuschko) [inline subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/samples/SamplesWebProjectIntegrationTest.groovy]: Yay! One down by just deleting the plugin.

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/samples/SamplesWebQuickstartIntegrationTest.groovy [hunk 4]
```diff
@@ -15,16 +15,13 @@
...
 import org.gradle.test.fixtures.file.LeaksFileHandles
 import org.gradle.test.fixtures.file.TestFile
 import org.gradle.util.ports.ReleasingPortAllocator
 import org.junit.Rule
-import spock.lang.Timeout
-import spock.lang.Unroll
-
-import static org.gradle.integtests.fixtures.UrlValidator.available
 
 @LeaksFileHandles
```
Comment (bmuschko) [inline subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/samples/SamplesWebQuickstartIntegrationTest.groovy]: We likely won't need this annotation anymore. It was only needed for the Jetty test case.

File: subprojects/docs/src/samples/webApplication/customized/build.gradle (hunk 1 above)
Comment (bmuschko) [inline subprojects/docs/src/samples/webApplication/customized/build.gradle]: Never mind...maybe we'll just keep it in and fix it if it breaks. It's good to have a working example that people can directly play around with.

**Comment by bmuschko** [comment]
Should also document the change in `notes.md` under "Changes to previously deprecated APIs". This section is introduced by https://github.com/gradle/gradle/pull/1726 so we should merge that first to avoid merge conflicts.

**Comment by bmuschko** [comment]
I noticed one more thing in https://docs.gradle.org/nightly/userguide/web_project_tutorial.html#sec:running_your_web_application.

> TODO: which url, configure port, uses source files in place and can edit your files and reload.

It was already there before your change but we should probably just remove this sentence.

**Comment by eriwen** [comment]
Good catch Ben. Fixed.

**Pull request merged**